src/tasks/withdraw/token_balances.ts

```diff
@@ -35,3 +64,36 @@ export async function getTokensWithBalanceAbove(
   }
   return result;
 }
+
+type BlockscoutAddressInfoResponse = BlockscoutSingleTokenInfo[];
+interface BlockscoutSingleTokenInfo {
+  token: {
+    address: string;
+    exchange_rate: string;
+    decimals: string;
+  };
+  value: string;
+}
+
+export async function getGnosisChainTokensWithBalanceAbove(
+  minValueUsd: string,
+  settlementContract: string,
+): Promise<string[]> {
+  const response = await axios.get(
+    `https://gnosis.blockscout.com/api/v2/addresses/${settlementContract}/token-balances`,
+  );
+  if (response.status !== 200) {
+    throw new Error(`Error getting tokens from Blockscout ${response}`);
+  }
+  const result = [];
+  const data = response.data as BlockscoutAddressInfoResponse;
+  for (const { value, token } of data) {
+    const tokenUsdValue =
+      parseFloat(token.exchange_rate) *
+      (parseInt(value) / Math.pow(10, parseInt(token.decimals)));
```

Comment:
I don't think I can avoid a conversion to float anywhere here. I don't know the precision of `exchange_rate`, so the best I can do is saying it's a float, which still offers some decent precision. Once this is a float, everything will need to be converted to float and so parsing as BigInt doesn't really help for precision. At first I tried to replace everything with `Number(BigInt(value) / 10n ** BigInt(token.decimals))`, which I realized was just going to create much worse issues!

Reply:
True. Unless we have something like a `BigRational` indeed the best option would be to use floats here. Doesn't look like ethers has something like that.
Also IIRC we end up getting quotes for all these tokens which should be more accurate anyway so the worst thing that could happen is that some of the tokens should get drained but doesn't. But then it will probably be drained the next time if the buffers keep accumulating that token.